bin/brew

```diff
@@ -41,149 +18,9 @@ HOMEBREW_PREFIX="$(chdir "$(dirname "$(dirname "$HOMEBREW_BREW_FILE")")" && pwd
-If you are using Xcode, you should:
-  sudo xcode-select -switch /Applications/Xcode.app
-Otherwise, you should:
-  sudo rm -rf /usr/share/xcode-select
-EOS
-  fi
-
-  XCRUN_OUTPUT="$(/usr/bin/xcrun clang 2>&1)"
-  XCRUN_STATUS="$?"
 
-  if [[ "$XCRUN_STATUS" -ne 0 && "$XCRUN_OUTPUT" = *license* ]]
-  then
-    odie <<EOS
-You have not agreed to the Xcode license. Please resolve this by running:
-  sudo xcodebuild -license
-EOS
-  fi
-fi
-
-# Many Pathname operations use getwd when they shouldn't, and then throw
-# odd exceptions. Reduce our support burden by showing a user-friendly error.
-if [[ ! -d "$(pwd)" ]]
-then
-  odie "The current working directory doesn't exist, cannot proceed."
-fi
-
-if [[ "$1" = -v ]]
-then
-  # Shift the -v to the end of the parameter list
-  shift
-  set -- "$@" -v
-fi
-
-HOMEBREW_COMMAND="$1"
-shift
-case "$HOMEBREW_COMMAND" in
-  ls)          HOMEBREW_COMMAND="list";;
-  homepage)    HOMEBREW_COMMAND="home";;
-  -S)          HOMEBREW_COMMAND="search";;
-  up)          HOMEBREW_COMMAND="update";;
-  ln)          HOMEBREW_COMMAND="link";;
-  instal)      HOMEBREW_COMMAND="install";; # gem does the same
-  rm)          HOMEBREW_COMMAND="uninstall";;
-  remove)      HOMEBREW_COMMAND="uninstall";;
-  configure)   HOMEBREW_COMMAND="diy";;
-  abv)         HOMEBREW_COMMAND="info";;
-  dr)          HOMEBREW_COMMAND="doctor";;
-  --repo)      HOMEBREW_COMMAND="--repository";;
-  environment) HOMEBREW_COMMAND="--env";;
-  --config)    HOMEBREW_COMMAND="config";;
-esac
-
-if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]] ; then
-  HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"
-elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]] ; then
-  HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
-fi
-
-if [[ "$(id -u)" = "0" && "$(/usr/bin/stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]]
-then
-  case "$HOMEBREW_COMMAND" in
-    install|reinstall|postinstall|link|pin|update|update-ruby|upgrade|create|migrate|tap|switch)
-      odie <<EOS
-Cowardly refusing to 'sudo brew $HOMEBREW_COMMAND'
-You can use brew with sudo, but only if the brew executable is owned by root.
-However, this is both not recommended and completely unsupported so do so at
-your own risk.
-EOS
-      ;;
-  esac
-fi
-
-if [[ -n "$HOMEBREW_BASH_COMMAND" ]]
-then
-  # source rather than executing directly to ensure the entire file is read into
-  # memory before it is run. This makes running a Bash script behave more like
-  # a Ruby script and avoids hard-to-debug issues if the Bash script is updated
-  # at the same time as being run.
-  #
-  # Hide shellcheck complaint:
-  # shellcheck source=/dev/null
-  source "$HOMEBREW_BASH_COMMAND"
-  { "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }
-else
-  exec "$HOMEBREW_RUBY_PATH" -W0 "$HOMEBREW_LIBRARY/brew.rb" "$HOMEBREW_COMMAND" "$@"
-fi
+exec /bin/bash "$HOMEBREW_LIBRARY/brew.sh" "$@"
```

Comment:
I think using `source "$HOMEBREW_LIBRARY/brew.sh"` should give the same effect with less overhead.